// Copyright 2021 The Pigweed Authors
//
// Licensed under the Apache License, Version 2.0 (the "License"); you may not
// use this file except in compliance with the License. You may obtain a copy of
// the License at
//
//     https://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
// License for the specific language governing permissions and limitations under
// the License.

syntax = "proto3";

package pw_software_update;

import "pw_software_update/tuf.proto";

message UpdateBundle {
  optional SignedMetadata root_metadata = 1;
  optional SignedMetadata timestamp_metadata = 2;
  optional SignedMetadata snapshot_metadata = 3;
  map<string, SignedMetadata> targets_metadata = 4;

  // Map of target path to target payload bytes.
  map<string, bytes> target_payload = 5;
}

// Update bundle metadata
// Designed to inform the update server what the device currently has in-place.
message Manifest {
  optional Metadata snapshot_metadata = 1;
  map<string, Metadata> targets_metadata = 2;
}
